### PR TITLE
Module 3

### PR DESCRIPTION
**Pull request by SChandarana**
Very barebones solution whilst I catch up with project exercises but this contains all the necessary parts

Planning to add the following when I have time:

- [ ] More comprehensive integration tests
- [ ] Selenium e2e testing
- [ ] Functionality stretch goals (e.g. completed task hiding etc)

### CODE REVIEW

File: todo_app/data/trello_items.py [hunk 1]
```diff
@@ -1,88 +1,79 @@
 import os
+from pyparsing import col
```
Comment (BenRamchandani) [inline todo_app/data/trello_items.py]: I don't think this is used

File: todo_app/data/trello_items.py [hunk 2]
```diff
@@ -1,88 +1,79 @@
...
-        }
+        params=generate_params({'name': item,
+                                'idList': column_id})
     )
 
+
 def move_item_to_column(item_id, column_name):
     column_id = get_column_id(column_name)
 
     url = (f'https://api.trello.com/1/cards/{item_id}')
```
Comment (BenRamchandani) [inline todo_app/data/trello_items.py]: Being pedantic, this allows query string injection with e.g. `curl -X 'POST' 'http://localhost:5000/complete_item/60a4c79a6c47a13f6e02e02c%3Fdesc%3DSneaky_description'`

```suggestion
    url = (f'https://api.trello.com/1/cards/{urllib.parse.quote(item_id, safe="")}')
```

**Review by BenRamchandani — APPROVED** [review]
Looks good, certainly happy to pass on Aptem :+1:

Couple of minor comments below. You've said you're planning to, but the Selenium stretch goal is worth doing if you have time as it makes some of the later project exercises slightly more involved. You can copy the template from <https://corndelwithsoftwire.github.io/DevOps-Course-FAQs/Modules/Module_3/Project_Exercise/code_samples.html#stretch-goal---selenium> - not sure if the version of the project exercise you has links there yet.

File: todo_app/data/trello_items.py (hunk 1 above)
Comment (SChandarana) [reply todo_app/data/trello_items.py]: Ah yep, classic auto-importing aha

File: todo_app/data/trello_items.py (hunk 2 above)
Comment (SChandarana) [reply todo_app/data/trello_items.py]: Ah interesting, yep that makes sense thanks! :D

**Pull request merged**